src/stream.c

```diff
@@ -471,9 +471,14 @@ static void rtp_handler(const struct sa *src, const struct rtp_header *hdr,
 			metric_inc_err(s->rx.metric);
 		}
 
-
-		if (stream_decode(s) == EAGAIN)
-			(void) stream_decode(s);
+		if (s->type == MEDIA_VIDEO) {
+			/* Read all ready frames */
+			while (stream_decode(s) == EAGAIN);
```

Comment:
Perhaps move semicolon to next line, to indicate that the while-loop is running a no-op.

Reply:
also consider a `for` loop instead of `while` loop to avoid potential deadlocks.
A high number can be chosen, e.g. in the range of hundreds ...